jwt/src/main/java/me/uport/sdk/jwt/JWTTools.kt

```diff
@@ -295,27 +312,50 @@ class JWTTools(
             PublicKey(pkBytes.toBigInteger()).normalize()
 
         }.filter { publicKey ->
+            try {
+                ecVerify(messageHash, sigData, publicKey)
+            } catch (ex: IllegalArgumentException) {
+                false
+            }
+        }
 
-            ecVerify(messageHash, sigData, publicKey)
+        fun hasEthereumAddressKeys(publicKeys: List<PublicKeyEntry>): Boolean {
+            return publicKeys.any { it.ethereumAddress != null }
+        }
+
+        if (matches.isEmpty() && hasEthereumAddressKeys(publicKeys)) {
+            return verifyRecoverableES256K(publicKeys, signatureBytes, signingInputBytes)
```

Comment:
fallback to the recoverable verification if there are `ethereumAddress` public keys available.

This is because the `ecVerify` method cannot work with `ethereumAddress`. It needs a full public key to verify against.
This means that a JWT could have been signed by the proper key but would be impossible to verify.